docs/README.md

```diff
@@ -404,6 +405,39 @@ Scan the directories were the Dockerfiles are to feed the scanner.
       - uses: Alfresco/alfresco-build-tools/.github/actions/docker-scan-image-dirs@ref
 ```
 
+### enforce-pr-conventions
+
+Checks if the branch name and pull request title follow conventions.
+
+```yaml
+      - uses: Alfresco/alfresco-build-tools/.github/actions/enforce-pr-conventions@ref
+        with:
+          jira-project-key: AAE|HXCS # optional
+          valid-branch-regex: "^(feature|test|tmp)\/AAE-[0-9]+-[A-Za-z0-9._-]+$" # optional
+          valid-pr-title-regex: "^AAE-[0-9]+ .+$" # optional
+```
+
+`jira-project-key`, `valid-branch-regex` and `valid-pr-title-regex` are optional.
+If `valid-branch-regex` and `valid-pr-title-regex` are not provided, the action will consume `jira-project-key` to generate the default regex.
+
+**Default regex for Branch name:**
+`"^(improvement|bug|feature|test|tmp)/(<jira-project-key>)-[0-9]+-[A-Za-z0-9._-]+$"`
+
+**Examples:**
+
+✅ improvement/AAE-12345-the-topic-of-the-branch
+
+❌ dev-uname-aae-12345
+
+**Default regex for PR title:**
+`"^(<jira-project-key>)-[0-9]+ .+$"`
+
+**Examples:**
+
+✅ AAE-12345 The title of the Merge Commit
+
+❌ AAE-12345 - The title of the Merge Commit
```

Comment:
Modified the regex accordingly.

Reply:
Actually looking at the regexp I'm not sure this is what we want: do we want to prevent people from starting their description with ":" or "-"?

If so maybe `^(<jira-project-key>)-[0-9]+ [A-Za-z0-9 ]+$"` should be replaced by "^(<jira-project-key>)-[0-9]+ [A-Za-z]{1}.*$"` (?) (regexp not tested)

Reply:
Yes, We want to prevent people from starting their description with '-' or ':' or any symbol. So it makes sense to use the suggested regex. Thanks!

Reply:
Have modified the regex.

Reply:
Thanks!